### PR TITLE
Basic emoji support

### PR DESCRIPTION
**Pull request by sendaimono**
Right now these emojis are supported:
:slightly_smiling_face:
:smile:
:stuck_out_tongue:
:kissing_heart:
:disappointed:
:confused:
:heart:
:kissing_heart:
:heart_eyes:

### CODE REVIEW

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            var transformedText = text
+            descriptors.forEach { descriptor: EmojiDescriptor->
+                var code: Int = whichEmoji(descriptor.content)
+                transformedText = transformedText.replace(descriptor.toTag(), String(Character.toChars(code)))
+            }
+            return transformedText
+        }
+
+        fun whichEmoji(text: String): Int {
+            val emojiPairs: MutableList<Pair<String, Int>> = mutableListOf((Pair("slightly_smiling_face", 0x1F600)),
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: you declare large list in loop 👎 
please move it outside the function to be decalred only once

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    Pair("disappointed", 0x1F641), Pair("smile", 0x1F603), Pair("stuck_out_tongue", 0x1F60B), Pair("kissing_heart", 0x1F618),
+                    Pair("confused", 0x1F615), Pair("heart", 0x2764), Pair("heart_eyes", 0x1F60D)
+            )
+            for (pair in emojiPairs) {
+                if (pair.first == text)
+                    return pair.second
+            }
+            return -1
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: use View.NO_ID constant

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    companion object {
+
+        fun detectEmojis(text: String): MutableList<EmojiDescriptor> {
+            val list: MutableList<EmojiDescriptor> = mutableListOf()
+            val pattern = Pattern.compile(":(.*?):")
+            val matcher = pattern.matcher(text)
+            while (matcher.find()) {
+                var group = matcher.group().replace(":","")
+                if (whichEmoji(group) != -1) {
+                    val thisEmojiDescriptor = EmojiDescriptor(group)
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: declare var before loop and overwrite it in loop

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package com.stfalcon.chatkit.messages
+
+import android.support.annotation.IntDef
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.util.regex.Pattern
+
+/**
+ * Created by Mikołaj on 2/15/2018.
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: modify header ;)

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageTextUtils.kt [hunk 5]
```diff
@@ -22,7 +22,7 @@ import java.util.regex.Pattern
 class MessageTextUtils {
 
 
-    companion object {
+        companion object {
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageTextUtils.kt]: why?

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageHolders.java [hunk 6]
```diff
@@ -557,8 +557,10 @@ public void applyStyle(MessagesListStyle style) {
     }
 
     public static void applyTextTransformation(TextView textView, String text, int gravity, MessagesListStyle style){
-        List<MessageTextUtils.PatternDescriptor> textUrls = MessageTextUtils.Companion.getTextPatterns(text);
-        @SuppressLint("RtlHardcoded") SpannableString sText = MessageTextUtils.Companion.transform(text, textUrls,
+        List<EmojisDetect.EmojiDescriptor> emojisDescriptors = EmojisDetect.Companion.detectEmojis(text);
+        String emojiText = EmojisDetect.Companion.transformText(text,emojisDescriptors);
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageHolders.java]: rename it to text, we dont want to have multiple instances of same text with different names

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            return transformedText
+        }
+
+        fun whichEmoji(text: String): Int {
+            val emojiPairs: MutableList<Pair<String, Int>> = mutableListOf((Pair("slightly_smiling_face", 0x1F600)),
+                    Pair("disappointed", 0x1F641), Pair("smile", 0x1F603), Pair("stuck_out_tongue", 0x1F60B), Pair("kissing_heart", 0x1F618),
+                    Pair("confused", 0x1F615), Pair("heart", 0x2764), Pair("heart_eyes", 0x1F60D)
+            )
+            for (pair in emojiPairs) {
+                if (pair.first == text)
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: brakets pls

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt (hunk 1 above)
Comment (m0d) [reply chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: and one Pair for one line

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageTextUtils.kt (hunk 5 above)
Comment (sendaimono) [reply chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageTextUtils.kt]: idk, some random tab press, didn't see that

File: chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt [hunk 8]
```diff
@@ -1,53 +1,64 @@
...
+            Pair("disappointed", 0x1F61E),
+            Pair("smile", 0x1F603),
+            Pair("stuck_out_tongue", 0x1F60B),
+            Pair("kissing_heart", 0x1F618),
+            Pair("confused", 0x1F615),
+            Pair("heart", 0x2764),
+            Pair("heart_eyes", 0x1F60D)
+    )
+
+    fun detectEmojis(text: String): MutableList<EmojiDescriptor> {
```
Comment (m0d) [inline chatkit/src/main/java/com/stfalcon/chatkit/messages/EmojisDetect.kt]: why all functions are removed from companion?

**Pull request merged**